### PR TITLE
[FIX][10.0] project_recalculate fix api.one & Redefining

### PR DESCRIPTION
**Pull request by tbaden**
fix api.one deprecated
fix Redefining built-in 'Warning'
fix Redefining built-in 'next'

### CODE REVIEW

File: project_recalculate/models/project_task.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 from odoo import models, fields, api, _
-from odoo.exceptions import Warning, ValidationError
+from odoo import exceptions
```
Comment (pedrobaeza) [inline project_recalculate/models/project_task.py:8]: Put it on the first row

File: project_recalculate/models/project_task.py [hunk 2]
```diff
@@ -174,12 +174,14 @@ def _calculation_prepare(self):
         increment = self.project_id.calculation_type == 'date_begin'
         if increment:
             if not self.project_id.date_start:
-                raise Warning(_('Start Date field must be defined.'))
+                raise exceptions.Warning(
```
Comment (pedrobaeza) [inline project_recalculate/models/project_task.py]: Use `exceptions.UserError`

File: project_recalculate/models/project_task.py [hunk 3]
```diff
@@ -174,12 +174,14 @@ def _calculation_prepare(self):
...
+                    _('Start Date field must be defined.')
+                )
             project_date = from_string(self.project_id.date_start)
             days = self.from_days
         else:
             if not self.project_id.date:
-                raise Warning(_('End Date field must be defined.'))
+                raise exceptions.Warning(_('End Date field must be defined.'))
```
Comment (pedrobaeza) [inline project_recalculate/models/project_task.py]: Use `exceptions.UserError`

File: project_recalculate/models/project_task.py [hunk 4]
```diff
@@ -251,8 +253,9 @@ def task_recalculate(self):
...
-    @api.one
+    @api.multi
     def write(self, vals):
-        vals = self._dates_onchange(vals)
-        vals = self._estimated_days_prepare(vals)
-        return super(ProjectTask, self).write(vals)
+        for this in self:
+            vals = this._dates_onchange(vals)
+            vals = this._estimated_days_prepare(vals)
+            return super(ProjectTask, this).write(vals)
```
Comment (pedrobaeza) [inline project_recalculate/models/project_task.py]: You can't return inside a loop. You should discard the returned value and return always `True`.

File: project_recalculate/tests/test_project_project.py [hunk 5]
```diff
@@ -165,7 +165,7 @@ def test_project_recalculate_exceptions(self):
             ('pj_6', 'date_end', False, False),
             ('pj_7', 'date_end', '2015-08-01', False),
         )
-        with self.assertRaises(Warning):
+        with self.assertRaises(exceptions.Warning):
```
Comment (pedrobaeza) [inline project_recalculate/tests/test_project_project.py]: Use `exceptions.UserError`

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by tbaden** [comment]
thanks for the feedback. changed it.

**Comment by tbaden** [comment]
runbot/travis checks all green with the requested changes

**Review by pedrobaeza — APPROVED** [review]
Fast-tracking this

**Pull request merged**